Services/Simulations.cs

```diff
@@ -297,6 +309,15 @@ public async Task<Models.Simulation> MergeAsync(SimulationPatch patch)
                 }
             }
 
+            if (simulation.Enabled)
+            {
+                // Reset ActualStartTime
+                simulation.ActualStartTime = null;
+
+                // Delete statistics records on simulation start
+                await this.simulationStatistics.DeleteSimulationStatisticsAsync(simulation.Id);
```

Comment:
a simulation can be modified also with PUT, so this code won't catch all the cases when a simulation starts. 

There is also a scenario where PUT and PATCH are called multiple times and the app shouldn't delete the metrics every time. For instance PUT and PATCH can be used to modify a running simulation without changing the value of `Enabled`

I think you should move this code in the partitioning agent, and execute it while the partitions and the devices are being created.

Reply:
As we discussed offline, updated PUT to call delete as well. Refactored PATCH so it is clear when the simulation was modified for start/stop.